extensions-core/lookups-cached-global/src/main/java/io/druid/query/lookup/NamespaceLookupExtractorFactory.java

```diff
@@ -203,26 +201,19 @@ public LookupExtractor get()
       throw Throwables.propagate(e);
     }
     try {
-      if (!started) {
+      if (entry == null) {
         throw new ISE("Factory [%s] not started", extractorID);
       }
-      String preVersion = null, postVersion = null;
-      Map<String, String> map = null;
-      // Make sure we absolutely know what version of map we grabbed (for caching purposes)
-      do {
-        preVersion = manager.getVersion(extractorID);
-        if (preVersion == null) {
-          throw new ISE("Namespace vanished for [%s]", extractorID);
-        }
-        map = manager.getCacheMap(extractorID);
-        postVersion = manager.getVersion(extractorID);
-        if (postVersion == null) {
-          // We lost some horrible race... make sure we clean up
-          manager.delete(extractorID);
-          throw new ISE("Lookup [%s] is deleting", extractorID);
-        }
-      } while (!preVersion.equals(postVersion));
-      final byte[] v = StringUtils.toUtf8(postVersion);
+      CacheScheduler.CacheState cacheState = entry.getCacheState();
+      if (cacheState == CacheScheduler.NoCache.CACHE_NOT_INITIALIZED) {
+        throw new ISE("%s not yet initialized cache, extractorID = %s", entry, extractorID);
+      }
+      if (cacheState == CacheScheduler.NoCache.ENTRY_CLOSED) {
+        throw new ISE("%s disposed, extractorID = %s", entry, extractorID);
+      }
+      CacheScheduler.VersionedCache versionedCache = (CacheScheduler.VersionedCache) cacheState;
```

Comment:
It is documented in Javadoc of `CacheScheduler.Entry.getCacheState()`. However I've changed this part of code a bit to make it a bit safer (or at least look safer) against adding more enum constants to `CacheScheduler.NoCache`: checking `cacheState instanceof NoCache` now instead of comparing with two enum constants separately.